3) Modify ubuntu snp guest launch to generalize this across various linux distros

LakshmiSaiHarika opened the pull request:
Built on branch pointing to PR #24 

Updated to standardize the Ubuntu SNP setup and launch guest process to easily add support for the guest seed image creation, setup and launch SNP guest on on non-debian linux distros.

LakshmiSaiHarika committed:
snp.sh: snp-guest-key location and guest user name update

Updated to show the correct details for the guest SSH access.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: Use of --guest-name and --guest-port command option to launch a separate SNP guest

This supports launch of various SNP Guests at the same time using unique guest name and guest qemu port.

A separate SNP guest can be launched with the --guest-name and --guest-port command option as follows:
  ./snp.sh launch-guest --guest-name <user-guest-name> --guest-port <user-guest-port>

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: Update ubuntu guest image download to optimize the guest creation process

Updated the ubuntu guest download process to download it once and re-use the same image to quickly create multiple guests with the improved performance.
Update of the  guest image download structure in ubuntu also helps to  easily add support for the same in the other OS linux distros.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
Update snp docs to add --guest-name and --guest-port option usage

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: Update to modularize ubuntu guest seed image creation process

Modularized guest seed image creation in ubuntu to standardize this process across non-debian OS linux distros that has no support for cloud-localds utility supported in canonical distribution.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
snp.sh: Updated Ubuntu SNP guest launch to generalize across various OS

Generalize a way to get and install SNP guest kernel package(AMDSEV build), generalize initrd pattern to make it match across different OS linux distros(initrd/initramfs), and set the default ubuntu guest kernel append parameter.

Signed-off-by: Harika Nittala <lnittala@amd.com>

DGonzalezVillal reviewed (commented):
lgtm, just consider the comments I made on the other PR.
This could have all been the same PR IMO.

LakshmiSaiHarika commented:
Acutally, the other PR is for optimizing the guest creation process only, and I thought this could be helpful in case to to apply these commits in the coco branch without standardizing upstream ubuntu guest creation process

LakshmiSaiHarika commented:
> lgtm, just consider the comments I made on the other PR. This could have all been the same PR IMO.

1. Addressed Diego's comments in the other PR (PR #24 )
2. Updated this PR to address the PR #24  comments raised by Diego
